camel/tasks/task.py

```diff
@@ -0,0 +1,321 @@
+# =========== Copyright 2023 @ CAMEL-AI.org. All Rights Reserved. ===========
+# Licensed under the Apache License, Version 2.0 (the “License”);
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an “AS IS” BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+# =========== Copyright 2023 @ CAMEL-AI.org. All Rights Reserved. ===========
+
+import re
+from enum import Enum
+from typing import Callable, Dict, List, Literal, Optional, Union
+
+from pydantic import BaseModel
+
+from camel.agents import ChatAgent
+from camel.messages import BaseMessage
+from camel.prompts import TextPrompt
+
+from .task_prompt import TASK_DECOMPOSE_PROMPT, TASK_EVOLVE_PROMPT
+
+
+class TaskSate(str, Enum):
+    OPEN = "OPEN"
+    RUNNING = "RUNNING"
+    DONE = "DONE"
+    DELETED = "DELETED"
+
+    @classmethod
+    def states(cls):
+        return [s.value for s in cls]
+
+
+class Task(BaseModel):
+    """
+    Task is specific assignment that can be passed to a agent.
+
+    Attributes:
+        content: string content for task.
+        id: A unique id.
+        state: The state which should be OPEN, RUNNING, DONE or DELETED.
+        type: task type
+        parent: The parent task, None for root task.
+        subtasks: The childrent sub-tasks for the task.
+        result: The answer for the task.
+    """
+
+    content: str
+    """The content of the task."""
+
+    id: str
+    """An unique string identifier for the task. This should ideally be
+    provided by the provider/model which created the task."""
+
+    state: TaskSate = TaskSate.OPEN
+    """The task state.
+    """
+
+    type: Optional[str] = None
```

Comment:
How is the type used? If this is tied to a functionality of the Task class itself, like types are defined by the framework. We should make it as a enum class.

Reply:
It's not defined currently. the `type` maybe useful in the future version, so just put it here.